modules/sdk-coin-algo/src/algo.ts

```diff
@@ -188,14 +187,24 @@ export class Algo extends BaseCoin {
     };
   }
 
+  /** inheritdoc */
+  generateRootKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = seed ? new AlgoLib.KeyPair({ seed }) : new AlgoLib.KeyPair();
+    const keys = keyPair.getKeys();
+    if (!keys.prv) {
+      throw new Error('Missing prv in key generation.');
+    }
+    return { prv: keys.prv + keys.pub, pub: keys.pub };
+  }
+
   /**
    * Return boolean indicating whether input is valid public key for the coin.
    *
    * @param {String} pub the pub to be checked
    * @returns {Boolean} is it valid?
    */
   isValidPub(pub: string): boolean {
-    return AlgoLib.algoUtils.isValidAddress(pub);
+    return AlgoLib.algoUtils.isValidAddress(pub) || AlgoLib.algoUtils.isValidPublicKey(pub);
```

Comment:
Algo has a `isValidPub` method, that's the accurate one to use i think. I remember i used the other ones and they gave a falsy value. Maybe add it as an || ?

Reply:
is the method being modified